src/shared/telemetry/telemetryEvent.ts

```diff
@@ -5,39 +5,58 @@
 
 'use strict'
 
-import { MetadataEntry, MetricDatum, Unit } from './clienttelemetry'
+import { AwsContext } from '../awsContext'
+import { MetadataEntry, MetricDatum } from './clienttelemetry'
+import { Datum, TelemetryNamespace } from './telemetryTypes'
 
 const NAME_ILLEGAL_CHARS_REGEX = new RegExp('[^\\w+-.:]', 'g')
 const REMOVE_UNDERSCORES_REGEX = new RegExp('_', 'g')
 
-export interface Datum {
-    name: string
-    value: number
-    unit?: Unit
-    metadata?: Map<string, string>
-}
-
 export interface TelemetryEvent {
     namespace: string
     createTime: Date
     data?: Datum[]
 }
 
-export function toMetricData(array: TelemetryEvent[]): MetricDatum[] {
+export function toMetricData(
+    array: TelemetryEvent[],
+    trimmedAwsContext: Pick<AwsContext, 'getCredentialAccountId'>
```

Comment:
Fetching the account id is out of scope for `toMetricData`. Instead of passing in the context, consider fetching the account id in `DefaultTelemetryClient`, and passing it directly to `toMetricData`.

Reply:
Should the client also handle the creation of the metadata object? I'm thinking that fetching the data should coincide with the creation of the object.

Reply:
> Should the client also handle the creation of the metadata object?

No, that is still the responsibility of `toMetricData`.

> I'm thinking that fetching the data should coincide with the creation of the object.

They should coincide, it just shouldn't happen inside `toMetricData`. For example

```typescript
// in caller
const accountId = context.getCredentialAccountId()
const metricData = toMetricData(array, accountId)
```